### ISSUE TITLE
Define Graphics engine API

### ISSUE THREAD

**Issue by zaryc0** [elliotk123/2D_Game_Engine#2]
## Define Graphics Engine API
the graphics API is to abstract the complexities of renedeing to the screen from the end user,
a series of functions need to be defined that will allow the end user to have their objects 
rendered to the screen regardless of what it is.
examples of things to handle include:

- png 
- jpeg
- defined polygons

this should be abstracted from the user as a simple function call e.g 
void Draw (object);

**Comment by elliotk123** [comment]
Using the recommended SDL rust wrapper, getting build errors currently as it can't find the library, may have to try installing SDL system wise first

**Comment by elliotk123** [comment]
Once the re-factored demo is merged into Integration, I'm happy to close as this is now covered by the render-sync, compositor and system interface
